keps/sig-node/2371-cri-pod-container-stats/README.md

```diff
@@ -0,0 +1,762 @@
+|                              |N/A                   |container_spec_memory_reservation_limit_bytes   |N/A                             |cAdvisor              |CRI or N/A                 |
+|                              |N/A                   |container_spec_memory_swap_limit_bytes          |N/A                             |cAdvisor              |CRI or N/A                 |
+|                              |N/A                   |container_start_time_seconds                    |N/A                             |cAdvisor              |CRI or N/A                 |
+|                              |N/A                   |container_tasks_state                           |N/A                             |cAdvisor              |CRI or N/A                 |
+|                              |N/A                   |container_threads                               |N/A                             |cAdvisor              |CRI or N/A                 |
+|                              |N/A                   |container_threads_max                           |N/A                             |cAdvisor              |CRI or N/A                 |
+|                              |N/A                   |container_file_descriptors                      |N/A                             |cAdvisor              |CRI or N/A                 |
+|                              |N/A                   |container_last_seen                             |N/A                             |cAdvisor              |CRI or N/A                 |
+|                              |                      |                                                |                                |cAdvisor              |CRI or N/A                 |
+
+## Motivation
+
+We want to avoid using cAdvisor for container & pod level stats and move metric collection to the CRI implementation for the following reasons:
+
+* cAdvisor and metric dependency: CRI mission is not fully fulfilled - container runtime is not fully plugable 
+* It would be ideal to break the monolithic design of cAdvisor (i.e. cAdvisor currently needs to be aware of the underlying container runtimes) 
+* Duplicate stats are collected by both cAdvisor and the CRI runtime, which can lead to:
+	* Different information from different sources
+	* Confusion: Unclear origin of a given metric
+	* Performance degradations (increased CPU / Memory / etc) [xref][perf-issue]
+* Stats should be reported by the container runtime which knows behavior of the container/pod the best.
+* cAdvisor only supports runtimes that run processes on the host, not e.g. VM based runtime like Kata Containers.
+* cAdvisor only supports linux containers, not Windows ones.
+* cAdvisor has big list of external dependencies since it needs to support multiple container runtimes itself - leads to large support surface.
+
+[perf-issue]: https://github.com/kubernetes/kubernetes/issues/51798
+
+
+### Goals
```

Comment:
Where does Prometheus fit in here?

Reply:
In the summary `This KEP will eventually also aim to have the CRI implementation populate the container level stats needed for the `/metrics/cadvisor` endpoint (or deprecate that endpoint).`